### PR TITLE
Bugs fix about ObjectCacher

### PR DESCRIPTION
**Pull request by majianpeng**

### CODE REVIEW

**Comment by majianpeng** [comment]
@dillaman. Can you review this? Thanks!

File: src/osdc/ObjectCacher.cc [hunk 1]
```diff
@@ -1059,6 +1076,13 @@ int ObjectCacher::_readx(OSDRead *rd, ObjectSet *oset, Context *onfinish,
   map<uint64_t, bufferlist> stripe_map;  // final buffer offset -> substring
   bool dontneed = rd->fadvise_flags & LIBRADOS_OP_FLAG_FADVISE_DONTNEED;
 
+  /*
+   * WARNING: we can only meaningfully return ENOENT if the read request
+   * passed in a single ObjectExtent.  Any caller who wants ENOENT instead of
+   * zeroed buffers needs to feed single extents into readx().
+   */  
+  assert(oset->return_enoent && rd->extents.size() == 1);
```
Comment (dillaman) [inline src/osdc/ObjectCacher.cc]: ```assert(!oset->return_enoent || rd->extents.size() == 1);```

**Comment by dillaman** [comment]
@majianpeng Just wondering if you have any updates?

**Comment by majianpeng** [comment]
@dillaman. Because on holiday so delay update this. please review . Thanks!

**Comment by majianpeng** [comment]
@dillaman . Updated by your suggestion. Please review again, Thanks!

**Pull request merged**